code/datums/elements/pet_cult.dm

```diff
@@ -0,0 +1,127 @@
+#define PET_CULT_ATTACK 10
+#define PET_CULT_HEALTH 50
+
+/datum/element/cultist_pet
+	element_flags = ELEMENT_BESPOKE
+	argument_hash_start_idx = 2
+	///our pet cult icon's pathfile
+	var/pet_cult_icon
+	///our pet cult icon state
+	var/pet_cult_icon_state
+
+/datum/element/cultist_pet/Attach(datum/target, pet_cult_icon = 'icons/mob/simple/pets.dmi', pet_cult_icon_state)
+	. = ..()
+
+	if(!isliving(target))
+		return ELEMENT_INCOMPATIBLE
+
+	src.pet_cult_icon = pet_cult_icon
+	src.pet_cult_icon_state = pet_cult_icon_state
+
+	RegisterSignal(target, COMSIG_LIVING_CULT_SACRIFICED, PROC_REF(become_cultist))
+	RegisterSignal(target, COMSIG_MOB_LOGIN, PROC_REF(on_login))
+	RegisterSignal(target, COMSIG_ATOM_UPDATE_ICON_STATE, PROC_REF(on_icon_state_updated))
+	RegisterSignal(target, COMSIG_ATOM_UPDATE_OVERLAYS, PROC_REF(on_overlays_updated))
+
+/datum/element/cultist_pet/Detach(datum/target)
+	. = ..()
+	UnregisterSignal(target, list(
+		COMSIG_MOB_LOGIN,
+		COMSIG_LIVING_CULT_SACRIFICED,
+		COMSIG_ATOM_UPDATE_ICON_STATE,
+		COMSIG_ATOM_UPDATE_OVERLAYS,
+	))
+
+/datum/element/cultist_pet/proc/on_overlays_updated(mob/living/basic/source, list/overlays)
+	SIGNAL_HANDLER
+
+	if(isnull(source.mind) && (FACTION_CULT in source.faction)) //cult indicator we show for non sentient pets
+		var/image/cult_indicator = image(icon = 'icons/mob/simple/pets.dmi', icon_state = "pet_cult_indicator", layer = ABOVE_GAME_PLANE)
+		overlays += cult_indicator
+
+/datum/element/cultist_pet/proc/on_icon_state_updated(mob/living/basic/source)
+	SIGNAL_HANDLER
+
+	if(pet_cult_icon_state && (FACTION_CULT in source.faction))
+		source.icon_state = pet_cult_icon_state
+		source.icon_living = pet_cult_icon_state
+
+///turn into terrifying beasts
+/datum/element/cultist_pet/proc/become_cultist(mob/living/basic/source, list/invokers, datum/team)
+	SIGNAL_HANDLER
+
+	if(source.stat == DEAD)
+		return
+
+	if(FACTION_CULT in source.faction)
+		return STOP_SACRIFICE
+
+	source.mind?.add_antag_datum(/datum/antagonist/cult, team)
+	qdel(source.GetComponent(/datum/component/obeys_commands)) //if we obey commands previously, forget about them
+	source.melee_damage_lower = max(PET_CULT_ATTACK, source::melee_damage_lower)
+	source.melee_damage_upper = max(PET_CULT_ATTACK + 5, source::melee_damage_upper)
```

Comment:
Maybe `PET_CULT_ATTACK_UPPER`? Feels weird to have a define which isn't actually the max value.